core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -776,8 +808,12 @@ object Chunk {
     case x: VectorChunk[A] => x.classTag
   }
 
-  private final case class Arr[A](private val array: Array[A]) extends NonEmpty[A] {
-    implicit val classTag: ClassTag[A] = ClassTag(array.getClass.getComponentType)
+  private def arr[A](array: Array[A]): Arr[A] =
+    Arr(array, ClassTag(array.getClass.getComponentType))
+
+  private final case class Arr[A](private val array: Array[A], ct: ClassTag[A]) extends NonEmpty[A] {
```

Comment:
I would not make `Arr` a case class, because you don't need `ct` to be a field, because you already create `classTag` as a field.